instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/SpanSuppressors.java

```diff
@@ -89,21 +88,20 @@ public boolean shouldSuppress(Context parentContext, SpanKind spanKind) {
   }
 
   static class ByContextKey implements SpanSuppressor {
+    private static final boolean canSuppress = canSuppress();
     private final SpanSuppressor delegate;
-    private final Method shouldSuppressInstrumentation;
 
     ByContextKey(SpanSuppressor delegate) {
       this.delegate = delegate;
-      Method shouldSuppressInstrumentation;
+    }
+
+    private static boolean canSuppress() {
       try {
-        Class<?> instrumentationUtil =
-            Class.forName("io.opentelemetry.exporter.internal.InstrumentationUtil");
-        shouldSuppressInstrumentation =
-            instrumentationUtil.getDeclaredMethod("shouldSuppressInstrumentation", Context.class);
-      } catch (ClassNotFoundException | NoSuchMethodException e) {
-        shouldSuppressInstrumentation = null;
+        Class.forName("io.opentelemetry.api.internal.InstrumentationUtil");
```

Comment:
This check could let users use instrumenation api with older sdk, idk whether this is really needed.

Reply:
I don't think this is needed unless someone is trying to build a custom agent with an older SDK version than the current one. However there is very little harm to keep it as is if it makes it a safer change.

Reply:
instrumentation-api can be used without the agent

Reply:
Good point, in this case it's simpler to just leave it as-is, maybe add an extra comment to make it extra clear (I don't trust my future self to remember this).

Reply:
Just wanted to point out that when the code is obfuscated (which is the typical case in Android though it's probably not exclusive to Android) this class name will change unless some exceptions are made for it, which can cause `canSuppress()` to return `false` by mistake.

We currently have a [rule here](https://github.com/open-telemetry/opentelemetry-android/blob/76d461bc29aa0ed5c0f53aa144e8bd6dff80c7ac/instrumentation/okhttp/okhttp-3.0/library/consumer-rules.pro#L1) to prevent `io.opentelemetry.exporter.internal.InstrumentationUtil` from being obfuscated to avoid breaking this functionality. We were planning to remove it once reflection wasn't needed here, though if we keep this line, we might have to change the rule instead. It's not a huge deal because we're aware of it and can make sure it won't break, though I think generally speaking we should try to keep reflection to a minimum.

Reply:
I removed the `InstrumentationUtil` presence check.